src/Aspire.Hosting/ExecutableResourceBuilderExtensions.cs

```diff
@@ -33,7 +34,8 @@ public static IResourceBuilder<ExecutableResource> AddExecutable(this IDistribut
                           {
                               context.Args.AddRange(args);
                           }
-                      });
+                      })
+                      .WithLifeCycleCommands();
```

Comment:
Rather than doing this I think it might be better to subscribe to an event and then loop across all the resources and add the commands. Something like this:

```csharp
builder.Eventing.Subscribe<BeforeStartEvent>((@event, ct) => {
 ... loop over the app model and inject commands into the resources.
});
```

The reason is that you'll need to add these common commands to each custom resource (e.g. JS and Python resource types). By using an event you can add this feature to all resource types that are either containers, projects or executable based without needing to leak the command annotation types out of the `Aspire.Hosting` assembly.

Reply:
... although type visibility is probably not an issue since we will want to expose `ResourceCommandAnnotation` in .NET 9.0 to allow for extensibility for folks.

Reply:
If the annotations were added in an event just before the resource was started, there isn't an opportunity for the built-in commands to be removed. I don't know if that's valuable, but I think it's one of the things you brought up in the design meeting.